sdk/cosmos/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/core/MultiTenantContainerCosmosFactory.java

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.data.cosmos.core;
+
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.spring.data.cosmos.CosmosFactory;
+
+/**
+ * Example for extending CosmosFactory for Mutli-Tenancy at the container level
+ */
+public class MultiTenantContainerCosmosFactory extends CosmosFactory {
+
+    public String manuallySetDatabaseName;
+
+    public String manuallySetContainerName;
+
+    /**
+     * Validate config and initialization
+     *
+     * @param cosmosAsyncClient cosmosAsyncClient
+     * @param databaseName      databaseName
+     */
+    public MultiTenantContainerCosmosFactory(CosmosAsyncClient cosmosAsyncClient, String databaseName) {
```

Comment:
just curious, why we do not have an constructor accepting containerName? and also in Spring, we do not really use setter pattern?

Reply:
By default, container name is taken from either entity name, or containerName in @Container within entity if defined. So it need not be passed to the constructor by default (but we add the setter/getter here so it can be overridden and called polymorphically in the case of multi-tenant scenario). For databaseName, this has to be passed in, as that has to be defined in config.